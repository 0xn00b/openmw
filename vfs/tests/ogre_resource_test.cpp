#include <iostream>
#include <OgreRoot.h>
#include <OgreResourceGroupManager.h>

/*
  This isn't really a test of our implementation, but a test of using
  the Ogre resource system to find files. If the Ogre interface
  changes and you have to change this test, you will have to change
  the servers/ogre_vfs.cpp implementation equivalently.

 */

using namespace std;
using namespace Ogre;

ResourceGroupManager *gm;
String group;

void find(const std::string &fileName)
{
  cout << "\nFile: " << fileName << endl;

  if(!gm->resourceExists(group, fileName))
    {
      cout << "Does not exist\n";
      return;
    }

  DataStreamPtr data = gm->openResource(fileName, group);

  cout << "Size: " << data->size() << endl;
  cout << "First line: " << data->getLine() << "\n";
}

int main()
{
  // Disable logging
  new LogManager;
  Log *log = LogManager::getSingleton().createLog("");
  log->setDebugOutputEnabled(false);

  // Set up Ogre
  Root root("","","");

  root.addResourceLocation("./", "FileSystem", "General");

  gm = ResourceGroupManager::getSingletonPtr();
  group = gm->getWorldResourceGroupName();

  find("Makefile");
  find("ogre_resource_test.cpp");
  find("bleh");

  cout << "\nAll source files:\n";
  FileInfoListPtr list = gm->findResourceFileInfo(group, "*.cpp");
  FileInfoList::iterator it, end;
  it = list->begin();
  end = list->end();
  for(; it != end; it++)
    cout << "  " << it->filename << endl;
}
