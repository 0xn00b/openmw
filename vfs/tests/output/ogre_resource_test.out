
File: Makefile
Size: 828
First line: GCC=g++ -I../

File: ogre_resource_test.cpp
Size: 1437
First line: #include <iostream>

File: bleh
Does not exist

All source files:
  physfs_server_test.cpp
  dummy_test.cpp
  ogre_resource_test.cpp
  server_common.cpp
  dummy_vfs.cpp
  ogre_client_test.cpp
  ogre_server_test.cpp
